flag_groups_test.go

```diff
@@ -43,22 +43,25 @@ func TestValidateFlagGroups(t *testing.T) {
 
 	// Each test case uses a unique command from the function above.
 	testcases := []struct {
-		desc                        string
-		flagGroupsRequired          []string
-		flagGroupsOneRequired       []string
-		flagGroupsExclusive         []string
-		subCmdFlagGroupsRequired    []string
-		subCmdFlagGroupsOneRequired []string
-		subCmdFlagGroupsExclusive   []string
-		args                        []string
-		expectErr                   string
+		desc                                  string
+		flagGroupsRequired                    []string
+		flagGroupsOneRequired                 []string
+		flagGroupsExclusive                   []string
+		flagGroupsIfPresentThenRequired       []string
+		subCmdFlagGroupsRequired              []string
+		subCmdFlagGroupsOneRequired           []string
+		subCmdFlagGroupsExclusive             []string
+		subCmdFlagGroupsIfPresentThenRequired []string
+		args                                  []string
+		expectErr                             string
 	}{
 		{
 			desc: "No flags no problem",
 		}, {
-			desc:                "No flags no problem even with conflicting groups",
-			flagGroupsRequired:  []string{"a b"},
-			flagGroupsExclusive: []string{"a b"},
+			desc:                            "No flags no problem even with conflicting groups",
+			flagGroupsRequired:              []string{"a b"},
+			flagGroupsExclusive:             []string{"a b"},
+			flagGroupsIfPresentThenRequired: []string{"a b"},
```

Comment:
Can you add a test about using


```suggestion
			flagGroupsIfPresentThenRequired: []string{"a b", "b a"},
```

This one is similar to the required group that also exists.

Then, it's up to you to decide if it should be discouraged or not

Please also consider adding a test about

```suggestion
			flagGroupsIfPresentThenRequired: []string{"a b", "a c"},
```

Sane here, you should decide if you want to discourage it or not

Same about a test about

```suggestion
			flagGroupsIfPresentThenRequired: []string{"a b c", "a c b"},
```


```suggestion
			flagGroupsIfPresentThenRequired: []string{"a b c", "b c"},
```

I'm asking for all these tests because I'm unsure how it would be expected to work